thread/thread.h

```diff
@@ -362,17 +362,52 @@ namespace photon
         {
             return waitq::wait(timeout);
         }
+        template <typename LOCK, typename PRED,
+                  typename = typename std::enable_if<std::is_same<
+                      decltype(std::declval<PRED>()()), bool>::value>::type>
```

Comment:
as long as PRED is not typed as Timeout

Reply:
and enable_if_t is shorter

Reply:
if do not limit PRED return value type, even `enable_if` part is meaningless.

Once PRED is callable (means PRED()() has a type) is good enough.